packages/flutter_tools/lib/src/commands/assemble.dart

```diff
@@ -105,9 +105,9 @@ class AssembleCommand extends FlutterCommand {
         'files will be written. Must be either absolute or relative from the '
         'root of the current Flutter project.',
     );
-    argParser.addOption(kExtraGenSnapshotOptions);
-    argParser.addOption(kExtraFrontEndOptions);
-    argParser.addOption(kDartDefines);
+    argParser.addOption(kExtraGenSnapshotOptions, help: '??? Please update assemble.dart to document this ???');
```

Comment:
This can be replaced with  https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command.dart#L658 for the existing documentation, same with https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/runner/flutter_command.dart#L488

Reply:
That will change the actual arguments supported here, because right now they are using the constants from `build_system/targets/common.dart` rather than the ones from `FlutterOptions`, is that ok?

Reply:
FWIW, it's causing some tests to fail, though I suspect that just means those tests are broken.

Reply:
...or actually, the code is broken...

Reply:
Oh I see - the names are different then. In that case, they have the same description and meaning but different names

Reply:
the names `flutter assemble` is using here today are really weird. Are we sure this is intentional?

Reply:
It's somewhat intentional. We can certainly change the argument names, but since this is the primary integration point for the re-entrant build you'll need to update tool_backend.dart, flutter.gradle, macos_assemble.sh and xcode_backend.sh

Reply:
ah interesting.

Reply:
ok i filed https://github.com/dart-lang/args/issues/181 to allow us to get aliases for arguments, which would let us support these legacy names and new ones, and in the meantime I made the code support the old ones but made it easy to switch back to the new ones.